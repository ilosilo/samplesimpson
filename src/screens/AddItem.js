import * as React from 'react';
import { useState } from 'react';
import { StackActions } from '@react-navigation/native';
import { listItem, textColor, addButton, buttonTextColor } from '../config/color'
import { saveItem, getAllKeys } from '../config/Storage';
import {
    View,
    TextInput,
    StyleSheet,
    Text,
    TouchableOpacity,
    ScrollView
} from 'react-native';

const AddItem = ({route,  navigation }) => {
    const [name, setName] = useState('');
    const [job, setJob] = useState('');
    const [about, setAbout] = useState('');
    const [avatar, setAvatar] = useState('');

    const item = {
        id: '',
        name: '',
        avatar: '',
        job: '',
        about: ''
    }
    
    const saveItemTostorage = () => {
        item.name = name;
        item.avatar = avatar;
        item.job = job;
        item.about = about;

        getAllKeys().then((allKeys) => {

            var max = -1;
            allKeys.forEach(key => {
                if (parseInt(key.valueOf()) > parseInt(max.valueOf())) {
                    max = key.valueOf();
                }
            });
            max = parseInt(max)+parseInt(1);
            
            item.id =(max).toString();
            saveItem(item).then(() => {
                navigation.dispatch(StackActions.popToTop())
                let newItems = route.params.items;
                newItems.push(item),
                route.params.setRendItems(newItems),
                route.params.onRefresh()
            })
        })
    }




    return (
        <View>
            <ScrollView>
            <Text style={st.inputText}>Name Surname</Text>
            <TextInput style={st.input}
                onChangeText={text => setName(text)}></TextInput>

            <Text style={st.inputText}>Job Title</Text>
            <TextInput style={st.input}
                onChangeText={text => setJob(text)}></TextInput>

            <Text style={st.inputText}>About Him/Her</Text>
            <TextInput style={st.input}
                multiline
                numberOfLines={4}
                onChangeText={text => setAbout(text)}></TextInput>

            <Text style={st.inputText}>Image Link</Text>
            <TextInput style={st.input}
                onChangeText={text => setAvatar(text)}></TextInput>
                </ScrollView>

            <TouchableOpacity style={st.submitButton}
                onPress={() => saveItemTostorage()}>
                <Text style={st.buttonText}>Add Character</Text>
            </TouchableOpacity>
        </View>
    );
};

export default AddItem;

const st = StyleSheet.create({
    input: {
        width: '90%',
        margin: 10,
        borderBottomColor: addButton,
        borderBottomWidth: 2,
        borderColor: listItem,
        borderWidth: 1,
    },
    inputText: {
        marginLeft: 12,
        marginTop: 15,
        color: textColor
    },
    submitButton: {
        backgroundColor: addButton,
        width: 220,
        height: 50,
        alignSelf: 'center',
        justifyContent: 'center',
        alignItems: 'center',
        marginTop: 30,
        borderRadius: 30,
    },
    buttonText: {
        color: buttonTextColor,
        fontWeight: 'normal',
        fontSize: 20
    },
    addButtonTextStyle: {
        fontSize: 30,
        color: 'white',
    },
    textStyle: {
        color: textColor,
        marginLeft: 15,
        fontWeight: 'bold'
    }
});
