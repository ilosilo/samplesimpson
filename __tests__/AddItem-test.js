
 import 'react-native';
 import React from 'react';
 import AddItem from '../src/screens/AddItem';
 
 import renderer from 'react-test-renderer';
 
 test('renders correctly', () => {
    const tree = renderer.create(<AddItem />).toJSON();
    expect(tree).toMatchSnapshot();
  });
 