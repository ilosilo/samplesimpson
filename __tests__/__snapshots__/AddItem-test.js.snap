// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View>
  <RCTScrollView>
    <View>
      <Text
        style={
          Object {
            "color": "rgb(84, 75, 63)",
            "marginLeft": 12,
            "marginTop": 15,
          }
        }
      >
        Name Surname
      </Text>
      <TextInput
        onChangeText={[Function]}
        style={
          Object {
            "borderBottomColor": "rgb(242, 145, 145)",
            "borderBottomWidth": 2,
            "borderColor": "rgba(238, 196, 196, 0.7)",
            "borderWidth": 1,
            "margin": 10,
            "width": "90%",
          }
        }
      />
      <Text
        style={
          Object {
            "color": "rgb(84, 75, 63)",
            "marginLeft": 12,
            "marginTop": 15,
          }
        }
      >
        Job Title
      </Text>
      <TextInput
        onChangeText={[Function]}
        style={
          Object {
            "borderBottomColor": "rgb(242, 145, 145)",
            "borderBottomWidth": 2,
            "borderColor": "rgba(238, 196, 196, 0.7)",
            "borderWidth": 1,
            "margin": 10,
            "width": "90%",
          }
        }
      />
      <Text
        style={
          Object {
            "color": "rgb(84, 75, 63)",
            "marginLeft": 12,
            "marginTop": 15,
          }
        }
      >
        About Him/Her
      </Text>
      <TextInput
        multiline={true}
        numberOfLines={4}
        onChangeText={[Function]}
        style={
          Object {
            "borderBottomColor": "rgb(242, 145, 145)",
            "borderBottomWidth": 2,
            "borderColor": "rgba(238, 196, 196, 0.7)",
            "borderWidth": 1,
            "margin": 10,
            "width": "90%",
          }
        }
      />
      <Text
        style={
          Object {
            "color": "rgb(84, 75, 63)",
            "marginLeft": 12,
            "marginTop": 15,
          }
        }
      >
        Image Link
      </Text>
      <TextInput
        onChangeText={[Function]}
        style={
          Object {
            "borderBottomColor": "rgb(242, 145, 145)",
            "borderBottomWidth": 2,
            "borderColor": "rgba(238, 196, 196, 0.7)",
            "borderWidth": 1,
            "margin": 10,
            "width": "90%",
          }
        }
      />
    </View>
  </RCTScrollView>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    nativeID="animatedComponent"
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "alignSelf": "center",
        "backgroundColor": "rgb(242, 145, 145)",
        "borderRadius": 30,
        "height": 50,
        "justifyContent": "center",
        "marginTop": 30,
        "opacity": 1,
        "width": 220,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "rgb(255,255,255)",
          "fontSize": 20,
          "fontWeight": "normal",
        }
      }
    >
      Add Character
    </Text>
  </View>
</View>
`;
